src/expression.c

```diff
@@ -118,7 +118,12 @@ Expression *getRightThis(Loc loc, Scope *sc, AggregateDeclaration *ad,
                     {
                         //printf("rewriting e1 to %s's this\n", f->toChars());
                         n++;
+#ifdef IN_GCC
+                        if (n > 1)
+                            e1 = new VarExp(loc, f->vthis);
+#else
                         e1 = new VarExp(loc, f->vthis);
+#endif
```

Comment:
I need to revisit this and work out what is going on.  I'm just in the middle of running the testsuite on a glue-layer fix for this.  @WalterBright can you provide an answer for why getRightThis needs to do this?

What I'm seeing is that the frontend rewrites a long dotvar expr such as ``this->this->this`` into just a plain ``this``.  Which is garbage from the backend's POV, as it generating code for a var decl that does not exist in the current local scope.  So gdc has to try and undo what getRightThis has done and reconstruct it back to it's rightful dotvar component reference, which is ``this->this->this``.